### PR TITLE
update latest pypi tag to include extras

### PR DESCRIPTION
**Pull request by nvijayrania**
Making the latest-pypi-tag to pickup packages with extras as well.

### CODE REVIEW

File: scripts/azureml-assets/azureml/assets/environment/pin_package_versions.py [hunk 1]
```diff
@@ -16,7 +16,7 @@
 from azureml.assets.util import logger
 
 # Handles {{latest-pypi-version}} and {{latest-pypi-version:flags}}
-LATEST_PYPI_VERSION = re.compile(r"([^\"'\s]+)([=~]=)\{\{latest-pypi-version(?::(.+))?\}\}")
+LATEST_PYPI_VERSION = re.compile(r"([^\"'\s]+?)(?:\[.*\])?([=~]=)\{\{latest-pypi-version(?::(.+))?\}\}")
```
Comment (elesel) [inline scripts/azureml-assets/azureml/assets/environment/pin_package_versions.py:19]: There wouldn't ever be a need to match `[]`
```suggestion
LATEST_PYPI_VERSION = re.compile(r"([^\"'\s]+?)(?:\[.+\])?([=~]=)\{\{latest-pypi-version(?::(.+))?\}\}")
```

**Comment by nvijayrania** [comment]
@elesel Thanks for taking a look at it. Could you please share the release process for the azureml-assets package. Actually I haven't worked with it before. Is there any release cadence with which this change can be released?

**Comment by elesel** [comment]
@nvijayrania The release process is automatic. Please update those two files and when the PR is merged the package will be updated in PyPI.

**Comment by nvijayrania** [comment]
Thanks @elesel : updated the PR.

File: scripts/azureml-assets/CHANGELOG.md [hunk 2]
```diff
@@ -3,6 +3,10 @@
 
 ### 🐛 Bugs Fixed
 
+## 1.16.35 (2023-03-05)
+### 🐛 Bugs Fixed
+- [#2407](https://github.com/Azure/azureml-assets/pull/2282) Add support for extra in LATEST_PYPI_VERSION env var
```
Comment (elesel) [inline scripts/azureml-assets/CHANGELOG.md:8]:
```suggestion
## 1.16.36 (2023-03-05)
### 🐛 Bugs Fixed
- [#2282](https://github.com/Azure/azureml-assets/pull/2282) Add support for optional dependencies with `latest-pypi-version`

```

**Review by elesel — CHANGES_REQUESTED** [review]

**Review by elesel — APPROVED** [review]

**Review by rjaincc — APPROVED** [review]

**Pull request merged**